src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.cpp

```diff
@@ -0,0 +1,369 @@
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Allocating Push Transport");
+
+        HandleCommand<Commands::AllocatePushTransport::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleAllocatePushTransport(ctx, commandData); });
+
+        break;
+
+    case Commands::DeallocatePushTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Deallocating Push Transport");
+
+        HandleCommand<Commands::DeallocatePushTransport::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleDeallocatePushTransport(ctx, commandData); });
+
+        break;
+
+    case Commands::ModifyPushTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Modifying Push Transport");
+
+        HandleCommand<Commands::ModifyPushTransport::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleModifyPushTransport(ctx, commandData); });
+
+        break;
+
+    case Commands::SetTransportStatus::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Setting Push Transport Status");
+
+        HandleCommand<Commands::SetTransportStatus::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleSetTransportStatus(ctx, commandData); });
+
+        break;
+
+    case Commands::ManuallyTriggerTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Manually Triggered Push Transport");
+
+        HandleCommand<Commands::ManuallyTriggerTransport::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleManuallyTriggerTransport(ctx, commandData); });
+
+        break;
+
+    case Commands::FindTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Finding Push Transport");
+
+        HandleCommand<Commands::FindTransport::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleFindTransport(ctx, commandData); });
+
+        break;
+    }
+}
+
+bool PushAvStreamTransportServer::FindStreamTransportConnection(const uint16_t connectionID)
+{
+    for (auto & id : mCurrentConnections)
+    {
+        if (id == connectionID)
+            return true;
+    }
+    return false;
+}
+
+uint16_t PushAvStreamTransportServer::GenerateConnectionID()
+{
+    static uint16_t assignedConnectionID = 0;
+    uint16_t nextConnectionID;
+
+    if (assignedConnectionID == MAX_PUSH_TRANSPORT_CONNECTION_ID)
+        nextConnectionID = 0;
+    else
+        nextConnectionID = assignedConnectionID + 1;
+
+    while (FindStreamTransportConnection(nextConnectionID) != false)
+    {
+        if (nextConnectionID == MAX_PUSH_TRANSPORT_CONNECTION_ID)
+            nextConnectionID = 0;
+        else
+            nextConnectionID = nextConnectionID + 1;
+    }
+    return nextConnectionID;
```

Comment:
OK, but nothing changed assignedConnectionID, so this will always return the closest thing to 0, which I am not sure is the intent....

I would also suggest removing the duplication of overflow handling via something like:
```
do {
   uint16_t nextConnectionID;
   // Compute it.
   if (FindStreamTransportConnection(nextConnectionID) == false) {
       return nextConnectionID
   }
} while (true);
```

or so.